tests/Aspire.Hosting.Nats.Tests/AddNatsTests.cs

```diff
@@ -1,18 +1,108 @@
+
+        var nats = appBuilder.AddNats("nats");
+        Assert.Equal("Aspire.Hosting.ApplicationModel.UserSecretsParameterDefault", nats.Resource.PasswordParameter!.Default?.GetType().FullName);
+    }
+
+    [Fact]
+    public void AddNatsDoesNotAddGeneratedPasswordParameterWithUserSecretsParameterDefaultInPublishMode()
+    {
+        using var appBuilder = TestDistributedApplicationBuilder.Create(DistributedApplicationOperation.Publish);
+
+        var nats = appBuilder.AddNats("nats");
+
+        Assert.NotEqual("Aspire.Hosting.ApplicationModel.UserSecretsParameterDefault", nats.Resource.PasswordParameter!.Default?.GetType().FullName);
+    }
+
+    [Fact]
+    public void AddNatsSetsDefaultUserNameAndPasswordOnNatsServerResource()
+    {
+        using var appBuilder = TestDistributedApplicationBuilder.Create(DistributedApplicationOperation.Publish);
+
+        var nats = appBuilder.AddNats("nats");
+
+        Assert.NotNull(nats.Resource.PasswordParameter);
+        Assert.False(string.IsNullOrEmpty(nats.Resource.PasswordParameter!.Value));
+    }
+
+    [Fact]
+    public void AddNatsSetsUserNameAndPasswordOnNatsServerResource()
+    {
+        using var appBuilder = TestDistributedApplicationBuilder.Create(DistributedApplicationOperation.Publish);
+
+        var userParameters = appBuilder.AddParameter("user", "usr");
+        var passwordParameters = appBuilder.AddParameter("pass", "password");
+
+        var nats = appBuilder.AddNats("nats", userName: userParameters, password: passwordParameters);
+
+        Assert.NotNull(nats.Resource.UserNameParameter);
+        Assert.NotNull(nats.Resource.PasswordParameter);
+
+        Assert.Equal("usr", nats.Resource.UserNameParameter!.Value);
+        Assert.Equal("password", nats.Resource.PasswordParameter!.Value);
+    }
+
+    [Fact]
+    public async Task AddNatsCreatesConnectionStringWithDefaultUserAndPassword()
+    {
+        var appBuilder = DistributedApplication.CreateBuilder();
+
+        appBuilder.AddNats("nats")
+            .WithEndpoint("tcp", e => e.AllocatedEndpoint = new AllocatedEndpoint(e, "localhost", 4222));
+
+        using var app = appBuilder.Build();
+
+        var appModel = app.Services.GetRequiredService<DistributedApplicationModel>();
+
+        var natsResource = Assert.Single(appModel.Resources.OfType<NatsServerResource>());
+        var connectionStringResource = natsResource as IResourceWithConnectionString;
+        Assert.NotNull(connectionStringResource);
+        var connectionString = await connectionStringResource.GetConnectionStringAsync();
+
+        Assert.Equal($"nats://nats:{natsResource.PasswordParameter?.Value}@localhost:4222", connectionString);
+        Assert.Equal("nats://nats:{nats-password.value}@{nats.bindings.tcp.host}:{nats.bindings.tcp.port}", connectionStringResource.ConnectionStringExpression.ValueExpression);
+    }
+
+    [Fact]
+    public async Task AddNatsCreatesConnectionStringWithUserAndPassword()
```

Comment:
There's not much difference between this test and `AddNatsSetsUserNameAndPasswordOnNatsServerResource`. Can they be combined into a single test?